projects/storefrontlib/src/cms-components/cart/add-to-cart/add-to-cart.component.ts

```diff
@@ -75,10 +103,33 @@ export class AddToCartComponent implements OnInit, OnDestroy {
   protected setStockInfo(product: Product): void {
     this.quantity = 1;
     this.hasStock = Boolean(
-      product.stock && product.stock.stockLevelStatus !== 'outOfStock'
+      product.stock && product.stock?.stockLevelStatus !== 'outOfStock'
     );
-    if (this.hasStock && product.stock?.stockLevel) {
-      this.maxQuantity = product.stock.stockLevel;
+    if (this.hasStock) {
+      if (product.stock?.stockLevel === undefined) {
```

Comment:
I don't think we need line 109 and 116. Wasn't it not good enough as it was before?

A product that is in stock but does not have a stock level still display only 'In stock'. What are you trying to achieve?

Reply:
https://github.com/SAP/spartacus/pull/11978#discussion_r661796453

Reply:
https://github.com/SAP/spartacus/pull/11978#discussion_r666186296

?